### ISSUE TITLE
No timeout when OBS not started

### ISSUE THREAD

**Issue by Grimtron** [BarRaider/obs-websocket-dotnet#9]
This might be an oversight on my behalf, but if try to connect to an OBS server that is not running, the Connect() call never ends. I do get the Disconnected() event, but it seems like the application is stuck in Connect() forever.

I already set WSTimeout to 2 seconds, but this did not help either.

Any idea on what might be the problem?

**Comment by Grimtron** [comment]
As I needed the stable operation of the library, I did some investigation on this. 

The problem is caused by the SendRequest() method, line 320: tcs.Task.Wait();

I added a 2000 milliseconds timeout and checked the result of the Wait method. If it returns false, I throw the ErrorResponseException, too, like this:

bool resultTaskWait = tcs.Task.Wait(2000);

if (tcs.Task.IsCanceled || !resultTaskWait)
    throw new ErrorResponseException("Request canceled");

Would probably be prudent to introduce a property for the timeout, but for now I am glad everything works as expected.

Thank you for the great lib!
